code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -314,6 +314,9 @@
 	else
 		ma.icon_state = initial(icon_state)
 
+	if(update_vore_icon()) // VOREStation edit
+		ma.icon_state = update_vore_icon()

```

Comment:
Bad, calling the same proc twice in a row without need, waste of CPU. Store the value somewhere and check it like:
```
    var/vore_icon_state = update_vore_icon()
    if(vore_icon_state)
        ma.icon_state = vore_icon_state
```
Put //VOREStation Edit Start and end